dtfs-central-api/src/services/portal/facility-amendment.validateAmendmentIsComplete.service.test.ts

```diff
@@ -0,0 +1,235 @@
+      },
+      {
+        description: 'changeCoverEndDate is true but coverEndDate is not provided',
+        amendment: {
+          ...aCompletedPortalFacilityAmendment(),
+          changeCoverEndDate: true,
+          coverEndDate: null,
+          isUsingFacilityEndDate: false,
+          bankReviewDate: new Date(),
+          facilityEndDate: null,
+        },
+        expectedErrorMessage: 'changeCoverEndDate is true but cover end date section incomplete',
+      },
+      {
+        description: 'changeCoverEndDate is true, isUsingFacilityEndDate is false but bankReviewDate is not provided',
+        amendment: {
+          ...aCompletedPortalFacilityAmendment(),
+          changeCoverEndDate: true,
+          coverEndDate: getEpochMs(),
+          isUsingFacilityEndDate: false,
+          bankReviewDate: null,
+          facilityEndDate: null,
+        },
+        expectedErrorMessage: 'changeCoverEndDate is true but cover end date section incomplete',
+      },
+      {
+        description: 'changeCoverEndDate and isUsingFacilityEndDate are true but facilityEndDate is not provided',
+        amendment: {
+          ...aCompletedPortalFacilityAmendment(),
+          changeCoverEndDate: true,
+          coverEndDate: getEpochMs(),
+          isUsingFacilityEndDate: true,
+          facilityEndDate: null,
+          bankReviewDate: null,
+        },
+        expectedErrorMessage: 'changeCoverEndDate is true but cover end date section incomplete',
+      },
+      {
+        description: 'changeCoverEndDate is true but isUsingFacilityEndDate is not provided',
+        amendment: {
+          ...aCompletedPortalFacilityAmendment(),
+          changeCoverEndDate: true,
+          coverEndDate: getEpochMs(),
+          isUsingFacilityEndDate: null,
+          bankReviewDate: new Date(),
+          facilityEndDate: null,
+        },
+        expectedErrorMessage: 'changeCoverEndDate is true but cover end date section incomplete',
+      },
+      {
+        description: 'changeFacilityValue is false but value provided',
+        amendment: {
+          ...aCompletedPortalFacilityAmendment(),
+          changeFacilityValue: false,
+          value: 1000,
+        },
+        expectedErrorMessage: 'changeFacilityValue is false but value provided',
+      },
+      {
+        description: 'changeFacilityValue is true but a value has not been provided',
+        amendment: {
+          ...aCompletedPortalFacilityAmendment(),
+          changeFacilityValue: true,
+          value: null,
+        },
+        expectedErrorMessage: 'changeFacilityValue is true but a value has not been provided',
+      },
+      {
+        description: 'effectiveDate not provided',
+        amendment: {
+          ...aCompletedPortalFacilityAmendment(),
+          effectiveDate: undefined,
+        },
+        expectedErrorMessage: 'effectiveDate not provided',
+      },
+      {
+        description: 'eligibilityCriteria answers are false',
```

Comment:
Might be worth making this instead `eligibilityCriteria contains a false answer` and having 2 criteria, one true, one false?